packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx

```diff
@@ -176,6 +176,7 @@ export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLBu
     inline,
     onOptionClick,
     open,
+    placeholderVisible: !value && !!placeholder,
```

Comment:
If value is 0 in the combo box then this will return true.  Consider testing against undefined and null explicitly for value.